### ISSUE TITLE
Enums are broken for nifs

### ISSUE THREAD

**Issue by vesperusiak** [tomas-abrahamsson/gpb#51]
I've got proto like this:

message ProtocolMessageResponse
{
        optional int32 protocol_version = 1 [default = 1];

        required int32 client_number = 2;

        enum Error {
                WRONG_SERVER = 0;
        }
        optional Error error = 30;
...
}

i'm generating code with command like this:
erl -boot start_clean -pa ./deps/gpb/ebin -noshell -noinput +B -I/some-path -o-nif-cc ./some-path/src -o-erl ./some_path/src -o-hrl ./include -nif -strbin -s gpb_compile c /some-path/protobuf_webserv.proto

and compiling like this:
g++ -g -fPIC -Wall -O3 -I./some_path/src -I/some_path/include -I./some_path/src -o ./some_path/src/protobuf_webserv.nif.o -c ./some_path/src/protobuf_webserv.nif.cc

and i'm getting compilation errors:
protobuf_webserv.nif.cc: In function ‘int p_msg_ProtocolMessageResponse(ErlNifEnv*, ERL_NIF_TERM, WebservServerProtocol::ProtocolMessageResponse*)’: protobuf_webserv.nif.cc:12198: error: ‘ProtocolMessageResponse_Error_WRONG_SERVER’ was not declared in this scope

protobuf_webserv.nif.cc: In function ‘ERL_NIF_TERM u_msg_ProtocolMessageResponse(ErlNifEnv*, const WebservServerProtocol::ProtocolMessageResponse*)’: protobuf_webserv.nif.cc:22843: error: ‘ProtocolMessageResponse_Error_WRONG_SERVER’ was not declared in this scope

I use gpb  3.17.9

**Comment by tomas-abrahamsson** [comment]
I guess your `protobuf_webserv.proto` also contains some `package x;` directive or similar? I managed to reproduce the error with such a directive, but not without, will take a look at it.

**Comment by tomas-abrahamsson** [comment]
Hi, just pushed 3.20.1, which I believe fixes this issue with enums, nifs and packages, hence closing. Feel free to re-open if there's anything still wrong.

**Comment by vesperusiak** [comment]
Sadly it didn't fix the issue
I get the same errors..
protobuf_webserv.proto contains some package CamelCaseName; as you guessed.

**Comment by tomas-abrahamsson** [comment]
When I tried out the change for 3.20.1, I once forgot to clean the build dir, so the `.nif.cc` did not get re-generated. Could that be it?

In the 3.20.1 change, I also added a test case, which is compiling to NIF, encoding and decoding a message via that NIF. The test compiles the `.proto` below. (One difference is enum location: There is already another test for nifs with the enum inside the message def, so for this, I thought I'd increase test diversity moving the enum to outside the message):
```proto
    package p1.p2;
        enum ee {
            ee1 = 1;
            ee2 = 2;
        };
    message ntest2 {
        optional ee f1 = 1;
    }
```
How should this be changed to make the error appear again?

**Comment by tomas-abrahamsson** [comment]
I'm very sorry for the too long delay before returning to this. My fault entirely. I took a new look, trying to recreate this issue more exactly, below is what I found.

The `.proto` file:
```proto
package CamelCaseName;
message ProtocolMessageResponse {
  optional int32 protocol_version = 1 [default = 1];
  required int32 client_number = 2;

  enum Error {
    WRONG_SERVER = 0;
  }
  optional Error error = 30;
}
```
My findings:
* When using the flags `... -nif -strbin ...` to gpb, things compile fine
* When using the flags `... -nif -strbin -pkgs ...` to gpb, I get compilation errors like below:
```
      ... error: ‘CamelCaseName::CamelCaseName’ has not been declared
```

Is this the issue you're seeing too? If so, does it work to remove the `-pkgs` option, or does that cause other issues?
